src/Tema1-DianaNiculescu.txt

```diff
@@ -0,0 +1,114 @@
+                                                  XPATH LOCATORS
+
+
+1) Locatorul  //button[@onclick='swapCheckbox()'] are urmatoarele:
+- tagname = button
+- attribute = onclick
+- attribute value = swapCheckbox()
+- rol: identifica in pagina "Dynamic Controls"(https://the-internet.herokuapp.com/dynamic_controls) butonul "Remove"
+
+2) Locatorul //div[@id='checkbox'] are urmatoarele:
+- tagname = div
+- attribute = id
+- attribute value = checkbox
+- rol: identifica in pagina "Dynamic Controls"(https://the-internet.herokuapp.com/dynamic_controls) checkbox-ul "A checkbox"
+
+3) Locatorul //input[@class='form-control-file'] are urmatoarele :
+- tagname = inpus
+- attribute = class
+- attribute value = form-control-file
+- rol: identifica in pagina 'Upload and Download'(https://demoqa.com/upload-download) butonul 'Choose File'
+
```

Comment:
Pentru//input[@autocomplete='off'] aveai 2 rezultate pentru ca acelasi atribut se gasea si la alt locator. Se recomanda sa se foloseasca atribute unice atunci cand se face xpath, in cazul nostru id. id este unic, si atunci locatorul cel mai corect si rapid ar fi //input[@id='userName']. Iar daca ar fi sa ai nevoie de acest camp in automation atunci cel mai corect si mai sigur e sa-l identifici prin id direct (By.id).